### ISSUE TITLE
Refractor code to use functional components.

### ISSUE THREAD

**Issue by pranay-simejia** [anitab-org/open-source-programs-web#76]
### Is your feature request related to a problem? Please describe.
As mentioned in React Docs after react-hook support, one should prefer using functional components.

### Describe the solution you'd like
Convert all class based components to functional components.

### Describe alternatives you've considered
for example, componentdDidMount can be replaced by useEffect hook.

**Comment by pranay-simejia** [comment]
Can I work on this issue?

**Comment by Rahulm2310** [comment]
@pranay-simejia I don't think this should have a priority currently. But, yeah, please wait for a maintainer to confirm/approve this issue before starting to work on the fix. Thanks 🙂 

Cc: @isabelcosta

**Comment by isabelcosta** [comment]
@pranay-simejia , @Rahulm2310 is correct. Right now we are trying to get this project to the MVP (minimum viable product) status. You can find more information on documentation [on this zulip message](https://anitab-org.zulipchat.com/#narrow/stream/237907-open-source-progs/topic/Open.20Sessions/near/225236028).
However, how much effort do you think this requires? Also, could you check on Zulip if more community members feel like this is a priority?

**Comment by pranay-simejia** [comment]
Switching to functional components can be done within 2 days. Single person's job.
@isabelcosta

**Comment by jalajcodes** [comment]
> Switching to functional components can be done within 2 days. Single person's job.
> @isabelcosta

Yeah I think there's also a vscode extension to do this too😅 I can't remember its name 🤔

**Comment by ayan-biswas0412** [comment]
> > Switching to functional components can be done within 2 days. Single person's job.
> > @isabelcosta
> 
> Yeah I think there's also a vscode extension to do this too😅 I can't remember its name 🤔

i think @jalajcodes you are talking abou this https://marketplace.visualstudio.com/items?itemName=wix.glean

**Comment by shruti1421** [comment]
I have also created this same issue earlier. Here is the issue number #69. 
If you give confirmation for this issue, I would to like to work on this.

**Comment by Aaishpra** [comment]
@shruti1421 @pranay-simejia since we are currently focusing more on completion of MVP(minimum viable product) i don't think this issue is of much use at this moment (considering it will require change in a lot of files and can cause a blocker to other features as well as current PRs) this issue can be put to hold until we have enough features and workforce for managing the project

**Comment by ayan-biswas0412** [comment]
> @shruti1421 @pranay-simejia since we are currently focusing more on completion of MVP(minimum viable product) i don't think this issue is of much use at this moment (considering it will require change in a lot of files and can cause a blocker to other features as well as current PRs) this issue can be put to hold until we have enough features and workforce for managing the project

but I think it will not be a huge task but in future when the codebase will be large then converting this thing will be a great problem but currently it is doable I think 

cc. @isabelcosta

**Comment by ayan-biswas0412** [comment]
> @shruti1421 @pranay-simejia since we are currently focusing more on completion of MVP(minimum viable product) i don't think this issue is of much use at this moment (considering it will require change in a lot of files and can cause a blocker to other features as well as current PRs) this issue can be put to hold until we have enough features and workforce for managing the project

ok
